ws-client/src/stream.rs

```diff
@@ -32,30 +32,29 @@ use futures::{
 };
 use pin_project::pin_project;
 use std::{io::Error as IoError, pin::Pin, task::Context, task::Poll};
+use tokio::net::TcpStream;
 use tokio_util::compat::{TokioAsyncReadCompatExt, TokioAsyncWriteCompatExt};
 
 /// Stream to represent either a unencrypted or encrypted socket stream.
 #[pin_project(project = EitherStreamProj)]
-#[derive(Debug, Copy, Clone)]
-pub enum EitherStream<S, T> {
+#[derive(Debug)]
+pub enum EitherStream {
```

Comment:
I did this but I realized I can just used placeholder type such as `()` to keep the generics here not sure that we need the generics anyway.

Unless we plan to support any transport using `tokio rustls` :)